src/org/exist/xquery/InlineFunction.java

```diff
@@ -41,6 +42,7 @@
 	public final static QName INLINE_FUNCTION_QNAME = new QName("", "");
 	
 	private UserDefinedFunction function;
+	private Stack<FunctionCall> calls = new Stack<>();
```

Comment:
Do you really mean `Stack`? `ArrayDeque` is much cheaper and provides the same interface if you don't need synchronization.